packages/welldone-wallet/src/lib/welldone.ts

```diff
@@ -0,0 +1,328 @@
+    );
+    const bytes = transaction.encode();
+    return Buffer.from(bytes).toString("base64");
+  };
+
+  const cleanup = () => {
+    if (_state.account) {
+      delete _state.account;
+    }
+  };
+
+  const getAccounts = async () => {
+    return _state.account ? [{ accountId: _state.account.accountId }] : [];
+  };
+
+  const signOut = async () => {
+    cleanup();
+    emitter.emit("signedOut", null);
+  };
+
+  const setupEvents = () => {
+    if (_state.wallet) {
+      _state.wallet.on("dapp:accountsChanged", async (newAccountId) => {
+        logger.log("onAccountChange", newAccountId);
+        await signOut();
+      });
+
+      _state.wallet.on("dapp:chainChanged", async (rpc) => {
+        logger.log("onNetworkChange", rpc);
+
+        const networkId = rpc.split(":")[1] === "near" ? "mainnet" : "testnet";
+
+        if (options.network.networkId !== networkId) {
+          await signOut();
+
+          emitter.emit("networkChanged", { networkId: networkId });
+        }
+      });
+    }
+  };
+
+  return {
+    async signIn() {
+      const existingAccounts = await getAccounts();
+
+      if (existingAccounts.length) {
+        return existingAccounts;
+      }
+
+      if (_state.account) {
+        await signOut();
+      }
+
+      const account = await _getAccounts();
+
+      const accessKey = await _validateAccessKey({
+        accountId: account[0],
+        publicKey: account[1],
+      });
+
+      if (!accessKey) {
+        signOut();
+        throw new Error(
+          `Public key (${account[1]}) is not registered with the account '${account[0]}'.`
+        );
+      }
+
+      _state.account = {
+        accountId: account[0],
+        publicKey: account[1],
+      };
+
+      setupEvents();
+
+      return getAccounts();
+    },
+
+    signOut,
+
+    getAccounts,
+
+    async verifyOwner({ message }) {
+      logger.log("Welldone:verifyOwner", { message });
+
+      if (!_state.wallet) {
+        throw new Error("Wallet is not installed");
+      }
+
+      const account = _state.account;
+
+      if (!account) {
+        throw new Error("Wallet not signed in");
+      }
+
+      const accountId = account.accountId;
+      const pubKey = utils.PublicKey.fromString(account.publicKey);
+      const block = await _state.wallet.request("near", {
+        method: "block",
+        params: {
+          finality: "final",
+        },
+      });
+
+      const data = {
+        accountId,
+        message,
+        blockId: block.header.hash,
+        publicKey: Buffer.from(pubKey.data).toString("base64"),
+        keyType: pubKey.keyType,
+      };
+      const encoded = JSON.stringify(data);
+
+      const signed = await _state.wallet.request("near", {
+        method: "dapp:sign",
+        params: [encoded],
+      });
+
+      return {
+        ...data,
+        signature: Buffer.from(signed.signature).toString("base64"),
+      };
+    },
+
+    async signAndSendTransaction({ signerId, receiverId, actions }) {
```

Comment:
You should consider using the `signTransactions` function in `packages\wallet-utils\src\lib\sign-transactions.ts` when signing to avoid using these "helper function" like `_validateAccessKey`, `convertTransaction` and `convertActions`. All of the above functions are used in `signTransactions`. This way we avoid duplicating code. 

You will have to provide a `signer: Signer` object to the `signTransactions` function. You can check out the `nightly` package and see how they have done it.

Reply:
We implemented a [dapp:sendTransaction](https://docs.welldonestudio.io/wallet/developer-guide/provider-api/method#sendTransaction) method of signing before sending a transaction on its own inside the wallet. So we use it instead of using the `signTransactions` function.

Reply:
Hey, I changed the `signAndSendTransaction` method to using `signTransactions` function. But the `signAndSendTransactions` method still uses the `dapp:sendTransaction` method (the wallet signs and sends transactions) because we do not currently support signing multiple transactions. This code will change after the wallet has been updated.

Reply:
Hi @Yoon-Suji , I'm unable to create or import wallet as well. Getting stuck on "Create a wallet" screen
![image](https://user-images.githubusercontent.com/111346686/196677135-709b514a-088e-4c30-9648-d292e1afef56.png)

Tried it on two machines with Chrome Versions 106.0.5249.119 (Official Build) (64-bit) and 106.0.5249.103 (Official Build) (64-bit). The same goes for Brave browser so you might want to investigate. 

Small thing I've noticed with password rule text... It says "write more than 8 letters" instead of "write 8 or more characters" if you want to change that.

Reply:
Hi @DamirSQA , We are trying to reproduce the exceptional situation where the error occurs. 
So I would really appreciate it if you could send me the console log by right-clicking on the `Create a Wallet` screen and clicking the inspect button.
Thanks for reporting the issue.

Reply:
You're welcome, here is the log

![image](https://user-images.githubusercontent.com/111346686/196904132-f85caaf3-a347-4fa9-a62e-758e0b6ffd5f.png)

Reply:
Thanks, @DamirSQA Could you turn off all the other wallet extensions and try again?

Reply:
Removing extensions didn't make a difference.  On some devices it does work, on the others does not.